language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1624,6 +1624,29 @@ public static boolean isSelfClassSymbol(Symbol symbol, PositionedOperationContex
         return classSymbol.equals(varTypeSymbol);
     }
 
+    /**
+     * Check if the symbol is an object symbol with self as the name.
+     *
+     * @param symbol               Symbol
+     * @param nodeAtCursor         Node
+     * @return {@link Boolean} whether the symbol is a self object symbol.
+     */
+
+    public static boolean isSelfObjectSymbol(Symbol symbol, Node nodeAtCursor) {
+        Node currentNode = nodeAtCursor;
+        while (currentNode != null) {
+            if (currentNode.kind() == SyntaxKind.OBJECT_CONSTRUCTOR) {
+                break;
+            }
+            currentNode = currentNode.parent();
+        }
+        if (currentNode != null) {
```

Comment:
We can rewrite these conditions as below:
```ballerina
if (currentNode == null || currentNode.kind() != SyntaxKind.OBJECT_CONSTRUCTOR) {
    return false;
}

return symbol.getName().isPresent() && symbol.getName().get().equals(SELF_KW);
```

Reply:
Done